mysql-test/main/invoking_user.test

```diff
@@ -0,0 +1,31 @@
+# MDEV-30908 - Test INVOKING_USER() function
+#
+# Check that INVOKING_USER() returns the right user and host
+# matching the content of `mysql.user` table for the user
+# executing the stored procedure.
+
+-- source include/not_embedded.inc
+
+--echo # Create a stored procedure to call the different ways to retrieve the user
+--delimiter /
+CREATE PROCEDURE mysql.user_function_demo() SQL SECURITY DEFINER BEGIN
```

Comment:
Could just concatenate the 3 functions with a `;` separate and return that. The select from mysql.user isn't particularly adding to the test case.

If this was removed, small simplification - don't need `begin/end` or delimiter if its just single statement.

Not need to put into the `mysql` database. Default `test` database serves the same function.

Reply:
Note a VIEW is another way the function could be exposed and a test around that would be good.

Can you check if a procedure calling a select on the view also returns the right results?